src/main/java/com/griddynamics/cloud/learning/dal/repositories/ArticleRepository.java

```diff
@@ -0,0 +1,178 @@
+            String caption = resultSet.getString("caption");
+            String description = resultSet.getString("description");
+            String content = resultSet.getString("content");
+            double price = resultSet.getDouble("price");
+            long author_id = resultSet.getLong("author_id");
+            String currency = resultSet.getString("currency");
+            Timestamp date = resultSet.getTimestamp("date");
+
+            return Article.builder()
+                    .id(id)
+                    .caption(caption)
+                    .description(description)
+                    .content(content)
+                    .price(price)
+                    .authorId(author_id)
+                    .currency(currency)
+                    .date(date.toLocalDateTime())
+                    .build();
+
+        } catch (SQLException e) {
+            Logger logger = Logger.getLogger(ArticleRepository.class.getName());
+            logger.log(Level.SEVERE, e.getMessage(), e);
+        }
+
+        return null;
+    }
+
+    @Override
+    protected String getEntityTableName() {
+        return Article.ARTICLE_TABLE;
+    }
+
+    @Override
+    public void add(Article item) {
+        String query = "INSERT INTO " + getEntityTableName() + " VALUES(?,?,?,?,?,?,?,?);";
+
+        try (Connection connection = DBConnectionManager.getConnection();
+             PreparedStatement statement = connection.prepareStatement(query)) {
+
+            statement.setLong(1, item.getId());
+            statement.setString(2, item.getCaption());
+            statement.setString(3, item.getDescription());
+            statement.setString(4, item.getContent());
+            statement.setDouble(5, item.getPrice());
+            statement.setLong(6, item.getAuthorId());
+            statement.setString(7, item.getCurrency());
+            statement.setTimestamp(8, Timestamp.valueOf(item.getDate()));
+
+            statement.executeUpdate();
+
+        } catch (SQLException e) {
+            Logger logger = Logger.getLogger(ArticleRepository.class.getName());
+            logger.log(Level.SEVERE, e.getMessage(), e);
+        }
+
+    }
+
+    @Override
+    public void update(Article item, String columnName, String newValue) {
+
+        String query = "UPDATE " + getEntityTableName() + " SET " + columnName + " = \'" + newValue +
+                "\' WHERE id = " + item.getId() + ";";
+
+        try (Connection connection = DBConnectionManager.getConnection();
+             PreparedStatement statement = connection.prepareStatement(query)) {
+
+            statement.executeUpdate();
+
+        } catch (SQLException e) {
+            Logger logger = Logger.getLogger(ArticleRepository.class.getName());
+            logger.log(Level.SEVERE, e.getMessage(), e);
+        }
+    }
+
+    public Optional<List<Article>> getFree() {
+
+        String query = "SELECT * FROM " + getEntityTableName() + " WHERE price = 0.0;";
```

Comment:
I don't like it. Seems having a boolean flag is gonna be a more effective way. Let's have one more field, "free" for example